pkg/component/resourcemanager/resource_manager.go

```diff
@@ -635,6 +637,11 @@ func (r *resourceManager) ensureConfigMap(ctx context.Context, configMap *corev1
 	}
 
 	if r.values.TargetDiffersFromSourceCluster {
+		// sort system component tolerations for a stable output
+		slices.SortFunc(r.values.SystemComponentTolerations, func(a, b corev1.Toleration) int {
+			return cmp.Compare(a.Key, b.Key)
+		})
```

Comment:
Wouldn't it make more sense to ensure a stable order in `ExtractSystemComponentsTolerations` (i.e., in the place that provides the `SystemComponentTolerations` to the GRM package)?
From GRM's `Values` API, the `SystemComponentTolerations` is already a `[]string` which is ordered. It isn't a big deal, but it somehow feels wrong to manipulate this user input. Maybe there is an intention behind the order 😉.

Reply:
No strong opinion from my side - will move it. I did it in the GRM deployment mainly because this is the component which knows or actually claims the order.